dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp

```diff
@@ -343,8 +343,14 @@ bool DeltaValueSpace::flush(DMContext & context)
 
         /// Update delta tree
         if (new_delta_index)
+        {
             delta_index = new_delta_index;
 
+            // Indicate that the index with old epoch should not be used anymore.
+            // This is useful in disaggregated mode which will invalidate the delta index cache in RN.
+            delta_index_epoch += 1;
```

Comment:
Seems there is no synchronization between `DeltaValueSpace::createSnapshot` (for_update=false) and `DeltaValueSpace::flush`. Is there a chance that we still get an incorrect delta-index?

Reply:
Good catch. We should ensure the delta_index and delta_index_epoch are either updated, or either not updated.